src/library/searchquery.cpp

```diff
@@ -166,6 +172,46 @@ QString TextFilterNode::toSql() const {
     return concatSqlClauses(searchClauses, "OR");
 }
 
+CrateFilterNode::CrateFilterNode(const QSqlDatabase& database,
+                                 const QString& argument)
+    : m_database(database),
+      m_argument(argument) {
+    getTrackIds();
+}
+
+void CrateFilterNode::getTrackIds() {
+    FieldEscaper escaper(m_database);
+    QString escapedArgument = escaper.escapeString(kSqlLikeMatchAll + m_argument + kSqlLikeMatchAll);
+
+    QSqlQuery query(m_database);
+    QString queryString = CrateStorage::formatQueryForTrackIdsByCrateName(escapedArgument);
+    query.setForwardOnly(true);
+    query.prepare(queryString);
+
+    if (query.exec()) {
+        if (sDebug) {
+            qDebug() << "Crate filter query: " << queryString;
+        }
+        while (query.next()) {
+            m_trackIds.append(TrackId(query.value(0)));
+        }
+    }
+}
+
+bool CrateFilterNode::match(const TrackPointer& pTrack) const {
+    return m_trackIds.contains(pTrack->getId());
+}
+
+QString CrateFilterNode::toSql() const {
+    QStringList trackIds;
```

Comment:
This approach is inefficient and doesn't scale well. Please use a subselect similar to formatSubselectQueryForCrateTrackIds():
formatSubselectQueryForCrateTrackIdsByCrateNameLike(const QString& crateName)